drivers/hmis/app/graphql/types/hmis_schema/project.rb

```diff
@@ -178,6 +182,8 @@ def unit_types
 
     # TODO use dataloader
     def units(**args)
+      raise HmisErrors::ApiError, 'Access denied' unless current_permission?(entity: object, permission: :can_view_units)
+
       resolve_units(**args)
     end
 
```

Comment:
I wasn't sure whether to add any permission gating on `has_units`. Searching in the frontend code, this doesn't seem to be used, but maybe I am missing it.

Reply:
Hmm not sure if/where it's used, but I think it can remain as-is